main/src/main/java/cgeo/geocaching/utils/FilterUtils.java

```diff
@@ -38,29 +39,40 @@ public static void openFilterActivity(final Activity activity, final GeocacheFil
                 filteredList, true);
     }
 
-    public static <T extends Activity & FilteredActivity> boolean openFilterList(final T filteredActivity, final GeocacheFilterContext filterContext) {
+    public static <T extends AbstractActivity & FilteredActivity> boolean openFilterList(final T filteredActivity, final GeocacheFilterContext filterContext) {
         final List<GeocacheFilter> filters = new ArrayList<>(GeocacheFilter.Storage.getStoredFilters());
+        final boolean isFilterActive = filterContext.get().isFiltering();
 
         if (filters.isEmpty()) {
+            filteredActivity.showToast(R.string.cache_filter_storage_load_delete_nofilter_message);
+
+            if (isFilterActive) {
+                SimpleDialog.of(filteredActivity).setTitle(R.string.cache_filter_storage_clear_title)
+                        .setPositiveButton(null)
+                        .setNeutralButton(TextParam.id(R.string.cache_filter_storage_clear_button))
```

Comment:
Is there a specific reason why you chose the neutral button? To my understanding a single button should be right-aligned (normally the "OK" button, but that can have a different label). Would read something like this:
```
SimpleDialog.of(filteredActivity).setTitle(R.string.cache_filter_storage_clear_title)
        .setPositiveButton(TextParam.id(R.string.cache_filter_storage_clear_button))
        .show(() -> filteredActivity.refreshWithFilter(GeocacheFilter.createEmpty(filterContext.get().isOpenInAdvancedMode())));
```

Reply:
This was on purpose to have it on the same location as in the other regular dialog to avoid switching places for same function

Reply:
Which other regular dialog do you mean - selecting a filter? (Although it looks similar, this would be something different - select a value from a list of one or more entries, without using an action button, versus selecting "Clear" as action button.)

Reply:
The dialog on long click on the filter button to select the filter from the list

![Screenshot_20241023_223735_cgeo.jpg](https://github.com/user-attachments/assets/3a5fd219-ad05-4643-bb6a-8bdcb751d58b)

Reply:
But it is the same click, where I expect the other dialog...

Reply:
Ok, thanks - I'd seen only the variant without the "clear" button:

![image](https://github.com/user-attachments/assets/62cd6c15-28ad-4a6b-9d27-1cf93891435b)

(Seems to depend on having a filter stored, but using a different one.)

IMHO the label should be right-aligned (using "positive" button) in both cases for consistency with material design. Although I agree that we have some aligning issue in the dialog - indentation of dialog heading is different from content, but that's a separate issue.

Reply:
> Ok, thanks - I'd seen only the variant without the "clear" button:
> 

The clear button appears, if you have a filter active